src/bp-document/bp-document-functions.php

```diff
@@ -5043,3 +5045,71 @@ function bb_document_remove_orphaned_download() {
 		}
 	}
 }
+
+
+/**
+ * Run migration for document description from post table to media table.
+ *
+ * @since BuddyBoss [BBVERSION]
+ *
+ * @param int $paged Page number.
+ *
+ * @return void
+ */
+function bb_document_description_migration( $paged = 1 ) {
+	global $wpdb, $bp, $bb_background_updater;
+
+	if ( empty( $paged ) ) {
+		$paged = 1;
+	}
+
+	$per_page = (int) apply_filters( 'bb_update_migrate_document_description', 30 );
+	$offset   = ( ( $paged - 1 ) * $per_page );
+
+	$documents = $wpdb->get_results(
+		$wpdb->prepare(
+			"SELECT id, attachment_id FROM {$bp->document->table_name} ORDER BY id ASC LIMIT %d offset %d",
+			$per_page,
+			$offset
+		)
+	);
+
+	if ( ! empty( $documents ) ) {
+		$bb_background_updater->push_to_queue(
+			array(
+				'type'     => 'migration',
+				'group'    => 'bb_update_migrate_document_description',
+				'priority' => 4,
+				'callback' => 'bb_update_migrate_document_description',
+				'args'     => array( $documents, $paged ),
+			)
+		);
+		$bb_background_updater->save()->schedule_event();
+	}
```

Comment:
run a single query instead of background jobs
`UPDATE wp_bp_document
JOIN wp_posts ON wp_posts.ID = wp_bp_media.attachment_id
SET wp_bp_document.description = wp_posts.post_content
ORDER BY wp_posts.ID DESC;`